javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/AgentInstaller.java

```diff
@@ -160,7 +140,58 @@ public static ResettableClassFileTransformer installBytebuddyAgent(
 
     agentBuilder = customizeByteBuddyAgent(agentBuilder);
     log.debug("Installed {} instrumenter(s)", numInstrumenters);
-    return agentBuilder.installOn(inst);
+    ResettableClassFileTransformer resettableClassFileTransformer = agentBuilder.installOn(inst);
+    installComponentsAfterByteBuddy(componentInstallers);
+    return resettableClassFileTransformer;
+  }
+
+  private static void installComponentsBeforeByteBuddy(
+      Iterable<ComponentInstaller> componentInstallers) {
+    ClassLoader savedContextClassLoader = Thread.currentThread().getContextClassLoader();
+    try {
+      Thread.currentThread().setContextClassLoader(AgentInstaller.class.getClassLoader());
+      for (ComponentInstaller componentInstaller : componentInstallers) {
+        componentInstaller.beforeByteBuddyAgent();
+      }
+    } finally {
+      Thread.currentThread().setContextClassLoader(savedContextClassLoader);
+    }
+  }
+
+  private static void installComponentsAfterByteBuddy(
+      Iterable<ComponentInstaller> componentInstallers) {
+    /*
+     * java.util.logging.LogManager maintains a final static LogManager, which is created during class initialization.
+     *
+     * JMXFetch uses jre bootstrap classes which touch this class. This means applications which require a custom log
+     * manager may not have a chance to set the global log manager if jmxfetch runs first. JMXFetch will incorrectly
+     * set the global log manager in cases where the app sets the log manager system property or when the log manager
+     * class is not on the system classpath.
+     *
+     * Our solution is to delay the initialization of jmxfetch when we detect a custom log manager being used.
+     *
+     * Once we see the LogManager class loading, it's safe to start jmxfetch because the application is already setting
+     * the global log manager and jmxfetch won't be able to touch it due to classloader locking.
+     */
+
+    /*
+     * Similar thing happens with AgentTracer on (at least) zulu-8 because it uses OkHttp which indirectly loads JFR
+     * events which in turn loads LogManager. This is not a problem on newer JDKs because there JFR uses different
+     * logging facility.
+     */
+    boolean appUsingCustomLogManager = isAppUsingCustomLogManager();
+    if (AgentInitializer.isJavaBefore9WithJfr() && appUsingCustomLogManager) {
+      log.debug("Custom logger detected. Delaying Agent Tracer initialization.");
+      for (ComponentInstaller componentInstaller : componentInstallers) {
+        registerClassLoadCallback(
+            "java.util.logging.LogManager",
+            new InstallComponentAfterByteBuddyCallback(componentInstaller));
+      }
```

Comment:
I think nice to pass the list into `InstallComponentsAfterByteBuddyCallback`, consolidating into a single callback